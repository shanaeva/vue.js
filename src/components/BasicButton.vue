<template>
  <button
    v-font:uppercase
    class="basic-button aqa-basic-button"
    :disabled="disabled"
    @click="click"
  >
    <slot />
  </button>
</template>

<script lang="ts">
import { defineComponent } from 'vue';

export default defineComponent({
  name: 'BasicButton',
  props: {
    disabled: { type: Boolean, require: false },
  },
  emits: ['click'],
  methods: {
    click() {
      this.$emit('click');
    },
  },
});
</script>

<style scoped>
  .basic-button {
    max-height: 60px;
    width: 220px;
    padding: 16px 56px;
    color: #ffffff;
    background: #f65261;
    border: none;
    border-radius: 4px;
    cursor: pointer;
  }
  .basic-button:hover {
    background: #bb3440;
  }
  .basic-button:disabled {
    background: #9b9494;
  }
</style>
