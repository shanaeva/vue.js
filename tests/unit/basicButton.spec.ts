import { shallowMount } from '@vue/test-utils';
import BasicButton from '@/components/BasicButton.vue';

describe('BasicButton', () => {
  it('render', () => {
    const wrapper = shallowMount(BasicButton, { props: { disabled: false } });
    expect(wrapper.props().disabled).toBe(false);
  });
  it('render correctly', () => {
    const wrapper = shallowMount(BasicButton, { props: { disabled: false } });
    expect(wrapper.vm.$el).toMatchSnapshot();
  });
});
