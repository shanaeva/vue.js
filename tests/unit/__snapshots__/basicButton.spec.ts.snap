// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicButton render correctly 1`] = `
<button
  class="basic-button aqa-basic-button"
>
  
  
</button>
`;
